azurerm/data_source_public_ip.go

```diff
@@ -20,6 +20,11 @@ func dataSourceArmPublicIP() *schema.Resource {
 
 			"resource_group_name": resourceGroupNameForDataSourceSchema(),
 
+			"ip_version": {
+				Type:     schema.TypeString,
+				Optional: true,
```

Comment:
since this is a Data Source this needs to be `Computed` rather than `Optional` (since the information is retrieved but not setable)